app/main.py

```diff
@@ -1,8 +1,39 @@
+from __future__ import annotations
+
+
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
+
+
+def find_partner(person: Person, spouse_name: str, is_wife: bool) -> None:
+    spouse = Person.people.get(spouse_name)
+    if spouse:
+        if is_wife:
+            person.wife = spouse
+            spouse.husband = person
+        else:
+            person.husband = spouse
+            spouse.wife = person
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    created_person_list = \
+        [
+            Person(persona["name"], persona["age"])
+            for persona in people
+        ]
+
+    for persona in people:
+        person = Person.people.get(persona["name"])
+        if person:
+            if "wife" in persona and persona["wife"]:
+                find_partner(person, persona["wife"], is_wife=True)
```

Comment:
You could simplify the check for the key and obtain the required straight away. Use walrus operator to achieve that:
```suggestion
            if wife_name := persona.get("wife"):
                find_partner(person, wife_name, is_wife=True)
```
Apply this check in the second if condition as well.